### PR TITLE
Test Plan: Streaming structured events via gNMI 

### PR DESCRIPTION
**Pull request by renukamanavalan**
### Description of PR
Plan for testing streaming structured events via gNMI

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

### CODE REVIEW

File: tests/streaming-events-telemetry.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+# Setup configuration
+Require: Few established BGP sessions and all *expected* dockers running.
+
+# Test
+This test will simulate scenarios to get the event fired and verify the fired event. It would also test cacheing, timeout, heartbeat & stats.
+
+# Test - Event receiver tool
+To test end to end, we need a tool that receives via gNMI
+
+1) Build a gNMI client tool (Go is easier) as part of SONiC-gNMI.
```
Comment (ganglyu) [inline tests/streaming-events-telemetry.md:19]: sonic-gnmi has provided gnmi_get, gnmi_set and gnmi_cli

File: tests/streaming-events-telemetry.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+3) Publish 2 events.
+4) Send SIGSTOP to eventd process
+5) Publish 3 more events
+6) Pause a second
+7) Send SIGCONT to eventd process.
+8) Wait for revceiver to exit; To handle error case, set a max timeout of 1 second. 
+9) Expect receiver to receive all 5.
+10) Expect non zero value for latency_in_ms.
+  
+8)
```
Comment (ganglyu) [inline tests/streaming-events-telemetry.md:140]: typo?

File: tests/streaming-events-telemetry.md (hunk 1 above)
Comment (renukamanavalan) [reply tests/streaming-events-telemetry.md:19]: You are correct. But we can't use gnmi-get or set for on change. The gnmi-cli would work, but it does not have an option to save response only in a o/p file. So a custom tool, that can provide the JSON strings in o/p file will be helpful to parse the o/p for test purposes. Additionally we could customize this tool with options as needed for testing, which may not be required in a user facing gnmi-cli tool.
Moreover the tool's code is from open-config and hence any update has to be via patch only, which adds additional restrictions.
Writing this tool should be pretty straight forward. We could use openconfig code as reference.

File: tests/streaming-events-telemetry.md (hunk 2 above)
Comment (renukamanavalan) [reply tests/streaming-events-telemetry.md]: Yes. Removed.

**Review by ganglyu — APPROVED** [review]

**Review by zbud-msft — APPROVED** [review]

**Pull request merged**